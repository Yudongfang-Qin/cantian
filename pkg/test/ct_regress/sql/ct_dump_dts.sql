conn sys/Huawei@123@127.0.0.1:1611
-- DTS202102020GQ0NOP0F00
drop table if exists dump_file;
create table dump_file
(
    f1 int not null, 
    f2 real, 
    f3 number, 
    f4 TIMESTAMP WITH LOCAL TIME ZONE,
    f5 TIMESTAMP WITH TIME ZONE,
    f6 date, 
    f7 timestamp
) 
PARTITION BY RANGE(f1)
(
    PARTITION part1 values less than(10),
    PARTITION part2 values less than(20),
    PARTITION part3 values less than(30),
    PARTITION part4 values less than(MAXVALUE)
);

LOAD DATA INFILE './data/dump_file_33M.DAT'  INTO TABLE dump_file;
select count(*) from dump_file;
dump TABLE dump_file into file 'dump.DAT' FILE SIZE '8b';
dump TABLE dump_file into file 'dump.DAT' FILE SIZE '-3';
dump TABLE dump_file into file 'dump_14M.DAT' FILE SIZE ' 14M ';
truncate table dump_file;
LOAD DATA INFILE 'dump_14M.DAT'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_14M.DAT_1'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_14M.DAT_2'  INTO TABLE dump_file;
select count(*) from dump_file;
dump TABLE dump_file into file 'dump_16M.DAT' FILE SIZE '16M';
truncate table dump_file;
LOAD DATA INFILE 'dump_16M.DAT'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_16M.DAT_1'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_16M.DAT_2'  INTO TABLE dump_file;
select count(*) from dump_file;
dump TABLE dump_file into file 'dump_16.3M.DAT' FILE SIZE '16.3M';
truncate table dump_file;
LOAD DATA INFILE 'dump_16.3M.DAT'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_16.3M.DAT_1'  INTO TABLE dump_file;
LOAD DATA INFILE 'dump_16.3M.DAT_2'  INTO TABLE dump_file;
select count(*) from dump_file;
drop table if exists dump_file;

drop table if exists test_table;
drop table if exists "test_table_new";
create table test_table (a int ,b varchar(10));
insert into  test_table values(1,'cantian1');
insert into  test_table values(2,'cantian2');
insert into  test_table values(3,'cantian3');
insert into  test_table values(4,'cantian4');
insert into  test_table values(5,'cantian5');
insert into  test_table values(6,'cantian6');
insert into  test_table values(7,'cantian7');
insert into  test_table values(8,'cantian8');
insert into  test_table values(9,'cantian9');
insert into  test_table values(10,'cantian10');
commit;
dump table test_table INTO FILE "case_tbl_hash";
create table "test_table_new" ("a" int ,"b" varchar(10),"c" varchar(10));

load data infile "case_tbl_hash" into table "test_table_new" trailing columns(a,b) SET c ='good';
load data infile "case_tbl_hash" into table "test_table_new" trailing columns("a","b") SET c ='good';
load data infile "case_tbl_hash" into table "test_table_new" trailing columns("a","b") SET "c" ='good';
select count(*) from "test_table_new";
drop table if exists test_table;
drop table if exists "case_tbl_hash";
