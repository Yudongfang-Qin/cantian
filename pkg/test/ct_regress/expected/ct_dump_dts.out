

SQL> conn sys/Huawei@123@127.0.0.1:1611

connected.

SQL> -- DTS202102020GQ0NOP0F00
SQL> drop table if exists dump_file;

Succeed.

SQL> create table dump_file
  2 (
  3     f1 int not null, 
  4     f2 real, 
  5     f3 number, 
  6     f4 TIMESTAMP WITH LOCAL TIME ZONE,
  7     f5 TIMESTAMP WITH TIME ZONE,
  8     f6 date, 
  9     f7 timestamp
 10 ) 
 11 PARTITION BY RANGE(f1)
 12 (
 13     PARTITION part1 values less than(10),
 14     PARTITION part2 values less than(20),
 15     PARTITION part3 values less than(30),
 16     PARTITION part4 values less than(MAXVALUE)
 17 );

Succeed.

SQL> 
SQL> LOAD DATA INFILE './data/dump_file_33M.DAT'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
135585 rows have been committed.
144624 rows have been committed.
153663 rows have been committed.
162702 rows have been committed.
171741 rows have been committed.
180780 rows have been committed.
189819 rows have been committed.
198858 rows have been committed.
207897 rows have been committed.
216936 rows have been committed.
225975 rows have been committed.
235014 rows have been committed.
244053 rows have been committed.
253092 rows have been committed.
262131 rows have been committed.
271170 rows have been committed.
280209 rows have been committed.
289248 rows have been committed.
298287 rows have been committed.
301125 rows have been committed.

Complete the data load.
totally read rows: 301125
     ignored rows: 0
      loaded rows: 301125
   committed rows: 301125
       error rows: 0
        skip rows: 0
SQL> select count(*) from dump_file;

COUNT(*)            
--------------------
301125              

1 rows fetched.

SQL> dump TABLE dump_file into file 'dump.DAT' FILE SIZE '8b';
5000 rows dumped.
10000 rows dumped.
15000 rows dumped.
20000 rows dumped.
25000 rows dumped.
30000 rows dumped.
35000 rows dumped.
40000 rows dumped.
45000 rows dumped.
50000 rows dumped.
55000 rows dumped.
60000 rows dumped.
65000 rows dumped.
70000 rows dumped.
75000 rows dumped.
80000 rows dumped.
85000 rows dumped.
90000 rows dumped.
95000 rows dumped.
100000 rows dumped.
105000 rows dumped.
110000 rows dumped.
115000 rows dumped.
120000 rows dumped.
125000 rows dumped.
130000 rows dumped.
135000 rows dumped.
140000 rows dumped.
145000 rows dumped.
150000 rows dumped.
155000 rows dumped.
160000 rows dumped.
165000 rows dumped.
170000 rows dumped.
175000 rows dumped.
180000 rows dumped.
185000 rows dumped.
190000 rows dumped.
195000 rows dumped.
200000 rows dumped.
205000 rows dumped.
210000 rows dumped.
215000 rows dumped.
220000 rows dumped.
225000 rows dumped.
230000 rows dumped.
235000 rows dumped.
240000 rows dumped.
245000 rows dumped.
250000 rows dumped.
255000 rows dumped.
260000 rows dumped.
265000 rows dumped.
270000 rows dumped.
275000 rows dumped.
280000 rows dumped.
285000 rows dumped.
290000 rows dumped.
295000 rows dumped.
300000 rows dumped.
301125 rows dumped.

Dump TABLE successfully:
  301125 rows are totally dumped.

SQL> dump TABLE dump_file into file 'dump.DAT' FILE SIZE '-3';
CT-00601, [1:53]Sql syntax error: invalid file size value!
SQL> dump TABLE dump_file into file 'dump_14M.DAT' FILE SIZE ' 14M ';
5000 rows dumped.
10000 rows dumped.
15000 rows dumped.
20000 rows dumped.
25000 rows dumped.
30000 rows dumped.
35000 rows dumped.
40000 rows dumped.
45000 rows dumped.
50000 rows dumped.
55000 rows dumped.
60000 rows dumped.
65000 rows dumped.
70000 rows dumped.
75000 rows dumped.
80000 rows dumped.
85000 rows dumped.
90000 rows dumped.
95000 rows dumped.
100000 rows dumped.
105000 rows dumped.
110000 rows dumped.
115000 rows dumped.
120000 rows dumped.
125000 rows dumped.
130000 rows dumped.
135000 rows dumped.
140000 rows dumped.
145000 rows dumped.
150000 rows dumped.
155000 rows dumped.
160000 rows dumped.
165000 rows dumped.
170000 rows dumped.
175000 rows dumped.
180000 rows dumped.
185000 rows dumped.
190000 rows dumped.
195000 rows dumped.
200000 rows dumped.
205000 rows dumped.
210000 rows dumped.
215000 rows dumped.
220000 rows dumped.
225000 rows dumped.
230000 rows dumped.
235000 rows dumped.
240000 rows dumped.
245000 rows dumped.
250000 rows dumped.
255000 rows dumped.
260000 rows dumped.
265000 rows dumped.
270000 rows dumped.
275000 rows dumped.
280000 rows dumped.
285000 rows dumped.
290000 rows dumped.
295000 rows dumped.
300000 rows dumped.
301125 rows dumped.

Dump TABLE successfully:
  301125 rows are totally dumped.

SQL> truncate table dump_file;

Succeed.

SQL> LOAD DATA INFILE 'dump_14M.DAT'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
126553 rows have been committed.

Complete the data load.
totally read rows: 126553
     ignored rows: 0
      loaded rows: 126553
   committed rows: 126553
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_14M.DAT_1'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
126553 rows have been committed.

Complete the data load.
totally read rows: 126553
     ignored rows: 0
      loaded rows: 126553
   committed rows: 126553
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_14M.DAT_2'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
48019 rows have been committed.

Complete the data load.
totally read rows: 48019
     ignored rows: 0
      loaded rows: 48019
   committed rows: 48019
       error rows: 0
        skip rows: 0
SQL> select count(*) from dump_file;

COUNT(*)            
--------------------
301125              

1 rows fetched.

SQL> dump TABLE dump_file into file 'dump_16M.DAT' FILE SIZE '16M';
5000 rows dumped.
10000 rows dumped.
15000 rows dumped.
20000 rows dumped.
25000 rows dumped.
30000 rows dumped.
35000 rows dumped.
40000 rows dumped.
45000 rows dumped.
50000 rows dumped.
55000 rows dumped.
60000 rows dumped.
65000 rows dumped.
70000 rows dumped.
75000 rows dumped.
80000 rows dumped.
85000 rows dumped.
90000 rows dumped.
95000 rows dumped.
100000 rows dumped.
105000 rows dumped.
110000 rows dumped.
115000 rows dumped.
120000 rows dumped.
125000 rows dumped.
130000 rows dumped.
135000 rows dumped.
140000 rows dumped.
145000 rows dumped.
150000 rows dumped.
155000 rows dumped.
160000 rows dumped.
165000 rows dumped.
170000 rows dumped.
175000 rows dumped.
180000 rows dumped.
185000 rows dumped.
190000 rows dumped.
195000 rows dumped.
200000 rows dumped.
205000 rows dumped.
210000 rows dumped.
215000 rows dumped.
220000 rows dumped.
225000 rows dumped.
230000 rows dumped.
235000 rows dumped.
240000 rows dumped.
245000 rows dumped.
250000 rows dumped.
255000 rows dumped.
260000 rows dumped.
265000 rows dumped.
270000 rows dumped.
275000 rows dumped.
280000 rows dumped.
285000 rows dumped.
290000 rows dumped.
295000 rows dumped.
300000 rows dumped.
301125 rows dumped.

Dump TABLE successfully:
  301125 rows are totally dumped.

SQL> truncate table dump_file;

Succeed.

SQL> LOAD DATA INFILE 'dump_16M.DAT'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
135585 rows have been committed.
144624 rows have been committed.
144632 rows have been committed.

Complete the data load.
totally read rows: 144632
     ignored rows: 0
      loaded rows: 144632
   committed rows: 144632
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_16M.DAT_1'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
135585 rows have been committed.
144624 rows have been committed.
144632 rows have been committed.

Complete the data load.
totally read rows: 144632
     ignored rows: 0
      loaded rows: 144632
   committed rows: 144632
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_16M.DAT_2'  INTO TABLE dump_file;
9039 rows have been committed.
11861 rows have been committed.

Complete the data load.
totally read rows: 11861
     ignored rows: 0
      loaded rows: 11861
   committed rows: 11861
       error rows: 0
        skip rows: 0
SQL> select count(*) from dump_file;

COUNT(*)            
--------------------
301125              

1 rows fetched.

SQL> dump TABLE dump_file into file 'dump_16.3M.DAT' FILE SIZE '16.3M';
5000 rows dumped.
10000 rows dumped.
15000 rows dumped.
20000 rows dumped.
25000 rows dumped.
30000 rows dumped.
35000 rows dumped.
40000 rows dumped.
45000 rows dumped.
50000 rows dumped.
55000 rows dumped.
60000 rows dumped.
65000 rows dumped.
70000 rows dumped.
75000 rows dumped.
80000 rows dumped.
85000 rows dumped.
90000 rows dumped.
95000 rows dumped.
100000 rows dumped.
105000 rows dumped.
110000 rows dumped.
115000 rows dumped.
120000 rows dumped.
125000 rows dumped.
130000 rows dumped.
135000 rows dumped.
140000 rows dumped.
145000 rows dumped.
150000 rows dumped.
155000 rows dumped.
160000 rows dumped.
165000 rows dumped.
170000 rows dumped.
175000 rows dumped.
180000 rows dumped.
185000 rows dumped.
190000 rows dumped.
195000 rows dumped.
200000 rows dumped.
205000 rows dumped.
210000 rows dumped.
215000 rows dumped.
220000 rows dumped.
225000 rows dumped.
230000 rows dumped.
235000 rows dumped.
240000 rows dumped.
245000 rows dumped.
250000 rows dumped.
255000 rows dumped.
260000 rows dumped.
265000 rows dumped.
270000 rows dumped.
275000 rows dumped.
280000 rows dumped.
285000 rows dumped.
290000 rows dumped.
295000 rows dumped.
300000 rows dumped.
301125 rows dumped.

Dump TABLE successfully:
  301125 rows are totally dumped.

SQL> truncate table dump_file;

Succeed.

SQL> LOAD DATA INFILE 'dump_16.3M.DAT'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
135585 rows have been committed.
144624 rows have been committed.
147343 rows have been committed.

Complete the data load.
totally read rows: 147343
     ignored rows: 0
      loaded rows: 147343
   committed rows: 147343
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_16.3M.DAT_1'  INTO TABLE dump_file;
9039 rows have been committed.
18078 rows have been committed.
27117 rows have been committed.
36156 rows have been committed.
45195 rows have been committed.
54234 rows have been committed.
63273 rows have been committed.
72312 rows have been committed.
81351 rows have been committed.
90390 rows have been committed.
99429 rows have been committed.
108468 rows have been committed.
117507 rows have been committed.
126546 rows have been committed.
135585 rows have been committed.
144624 rows have been committed.
147343 rows have been committed.

Complete the data load.
totally read rows: 147343
     ignored rows: 0
      loaded rows: 147343
   committed rows: 147343
       error rows: 0
        skip rows: 0
SQL> LOAD DATA INFILE 'dump_16.3M.DAT_2'  INTO TABLE dump_file;
6439 rows have been committed.

Complete the data load.
totally read rows: 6439
     ignored rows: 0
      loaded rows: 6439
   committed rows: 6439
       error rows: 0
        skip rows: 0
SQL> select count(*) from dump_file;

COUNT(*)            
--------------------
301125              

1 rows fetched.

SQL> drop table if exists dump_file;

Succeed.

SQL> 
SQL> drop table if exists test_table;

Succeed.

SQL> drop table if exists "test_table_new";

Succeed.

SQL> create table test_table (a int ,b varchar(10));

Succeed.

SQL> insert into  test_table values(1,'cantian1');

1 rows affected.

SQL> insert into  test_table values(2,'cantian2');

1 rows affected.

SQL> insert into  test_table values(3,'cantian3');

1 rows affected.

SQL> insert into  test_table values(4,'cantian4');

1 rows affected.

SQL> insert into  test_table values(5,'cantian5');

1 rows affected.

SQL> insert into  test_table values(6,'cantian6');

1 rows affected.

SQL> insert into  test_table values(7,'cantian7');

1 rows affected.

SQL> insert into  test_table values(8,'cantian8');

1 rows affected.

SQL> insert into  test_table values(9,'cantian9');

1 rows affected.

SQL> insert into  test_table values(10,'cantian10');

1 rows affected.

SQL> commit;

Succeed.

SQL> dump table test_table INTO FILE "case_tbl_hash";
10 rows dumped.

Dump TABLE successfully:
  10 rows are totally dumped.

SQL> create table "test_table_new" ("a" int ,"b" varchar(10),"c" varchar(10));

Succeed.

SQL> 
SQL> load data infile "case_tbl_hash" into table "test_table_new" trailing columns(a,b) SET c ='good';
CT-00601, [1:8]Sql syntax error: invalid column name 'A'
SQL> load data infile "case_tbl_hash" into table "test_table_new" trailing columns("a","b") SET c ='good';
CT-00618, [1:72]The column 'C' was invalid
ZS-00003: worker init failed

Failure happens and loading process is interrupted.
totally read rows: 0
     ignored rows: 0
      loaded rows: 0
   committed rows: 0
       error rows: 0
        skip rows: 0
SQL> load data infile "case_tbl_hash" into table "test_table_new" trailing columns("a","b") SET "c" ='good';
10 rows have been committed.

Complete the data load.
totally read rows: 10
     ignored rows: 0
      loaded rows: 10
   committed rows: 10
       error rows: 0
        skip rows: 0
SQL> select count(*) from "test_table_new";

COUNT(*)            
--------------------
10                  

1 rows fetched.

SQL> drop table if exists test_table;

Succeed.

SQL> drop table if exists "case_tbl_hash";

Succeed.

SQL> 


